src/preload.py

```diff
@@ -609,8 +609,64 @@ def write_resin_device_pinning(app_data, output):
         )
 
 
+# Remove once supervisor is part of the hostapp
+def isSupervisor(app, images):
+    if len(app['services']) != 1:
+        # Supervisor is single service
+        return False
+
+    # Fetch image name from state app
+    for service in app['services']:
+        image_name = app['services'][service]['image']
+
+    for image in images:
+        if re.match(image, image_name):
+            return True
+    return False
+
+
 def write_apps_json(data, output):
-    """Writes data dict to output as json"""
+    """Updates or creates preloaded state"""
+    if os.path.isfile(output):
+        with open(output, "r") as f:
+            current_state = json.load(f)
+
+        # Check state version consistency
+        for uuid in data.keys():
+            if len(uuid) != 32 and len(uuid) != 62:
+                raise Exception("Refusing to preload v2 target state app")
```

Comment:
I was wondering about this while looking into how the app state is passed to the preloader code. The `preload.js` will check the supervisor version in the image and fetch a versioned target state for the app depending on the supervisor version, so there should never be a mismatch here.
This check should disappear as it will never be hit, or if we want to be paranoid check that the id length of both states are the same, and point the user to contact support if they are not.